csharp/SecureMemory/SecureMemory/SecureMemorySecretFactory.cs

```diff
@@ -180,11 +183,12 @@ private static IProtectedMemoryAllocator DetectViaRuntimeInformation(IConfigurat
             return null;
         }
 
-        private static IProtectedMemoryAllocator ConfigureForLinux64(IConfiguration configuration)
+        private static ISecureMemoryAllocator ConfigureForLinux64(IConfiguration configuration)
         {
             if (configuration != null)
             {
-                var secureHeapEngine = configuration["secureHeapEngine"];
+                string secureHeapEngine = configuration["secureHeapEngine"];
+                bool.TryParse(configuration["mlock"], out bool mLock);
```

Comment:
Need to ensure the use of `mlock` is the default behavior of this library, but this seems to require not only the use of a configuration (which should be optional) but the "mlock" config setting must be set to "true" to enable the `mlock` protection.

Once quick fix would be to use "enabled" and "disabled" (defaulting of "enabled" of course) in place of "true"/"false"